### PR TITLE
WIP: Save limit value in localStorage

### PR DESCRIPTION
**Pull request by polRk**
### Description
Put a separate option in the settings tab.
Save pageSize default value in localStorage.
Load page Size default value in Browser Rows view

### Affected components
- [x] Console

### Related Issues
#3849

### CODE REVIEW

**Comment by polRk** [comment]
@hasura-bot deploy

**Comment by rikinsk** [comment]
@polRk Thanks for taking this issue up and contributing. 

I just had some feedback on the UX. Instead of adding a new page in the setting section to set the default pageSize value, it would just suffice to persist the page size value when the user updates the page size on the browse rows page itself. If user sets page value to 20, then we persist that value and show 20 results always from then on.

Screenshot from 2020-02-11 12-57-05

**Comment by polRk** [comment]
@rikinsk Looks pretty good. I'll try to finish it soon

**Comment by polRk** [comment]
After update, i see 2 requests ( with limit 10, and limit from localStorage )
Снимок экрана 2020-02-11 в 3 38 34 PM

Снимок экрана 2020-02-11 в 3 38 24 PM

**Comment by polRk** [comment]
@beerose Can you make a review ?

**Comment by beerose** [comment]
I'll do the review first thing tomorrow morning.

File: console/src/components/Services/EventTrigger/PendingEvents/ViewActions.js [hunk 1]
```diff
@@ -379,7 +379,6 @@ const pendingEventsReducer = (triggerName, triggerList, viewState, action) => {
               where: { delivered: false, error: false },
             },
           ],
-          limit: 10,
```
Comment (beerose) [inline console/src/components/Services/EventTrigger/PendingEvents/ViewActions.js]: Don't delete this line.

**Comment by polRk** [comment]
@beerose How can i changed the base ?

**Comment by beerose** [comment]
@polRk You would need to create a new PR and choose my fork and `browse-table-improvements` branch. 

Screenshot 2020-02-12 at 12 51 07

But what happened here?

Screenshot 2020-02-12 at 12 54 01

**Comment by polRk** [comment]
> But what happened here?

I made a rebase to get new code from the master branch. Standard practice

**Comment by beerose** [comment]
> > But what happened here?
> 
> I made a rebase to get new code from the master branch. Standard practice

I think something went wrong here because you have 706 commits to merge. 
What you could do now is:
1. Fetch the newest Hasura master.
```
git remote add hasura git@github.com:hasura/graphql-engine.git

git fetch hasura master
```
2. Do a soft reset:
```
git reset --soft hasura/master
```
3. Commit your changes.
```
git commit -m 'Save limit value in localStorage'
```

**Comment by polRk** [comment]
> I think something went wrong here because you have 706 commits to merge.

I rebased wrong master. Solved, thank you

**Comment by beerose** [comment]
Hi @polRk!  It's been a while since your last commit, so I want to ask if you still plan to work on this PR?

**Comment by polRk** [comment]
Hi @beerose!  I don’t have enough time for this PR.  I want to make hasura console better, if someone can to do this faster than me, it will be good. Sorry for waiting

**Comment by beerose** [comment]
Closing in favor of https://github.com/hasura/graphql-engine/pull/4323.

**Pull request closed without merging**